src/algorithms/pathfinders/dijkstra.ts

```diff
@@ -0,0 +1,111 @@
+import EdgeCost from '../models/edgecost';
+import Graph from '../models/graph';
+import Vertex from '../models/vertex';
+/**
+ * Implements Dijkstra's algorithm to find the shortest path between 2 vertices
+ */
+export default class DijkstraPathfinder {
+    /**
+     * Stores best known distances from starting vertex to all the others
+     */
+    distances: Map<number, {

```

Comment:
When you want to use a specific object, like this example with distance and sourceID, it's a good practise to make a class for it or at least an interface. This would guarantee that what you're dealing with is really the desired object with given properties. (This is not a big issue at all)